examples/MomentoApplication/Program.cs

```diff
@@ -6,43 +6,94 @@
 using Momento.Sdk.Exceptions;
 using Momento.Sdk.Responses;
 
-namespace MomentoApplication
+string MOMENTO_AUTH_TOKEN = Environment.GetEnvironmentVariable("MOMENTO_AUTH_TOKEN");
+// Use a GUID for the cache name so it's unlikely to exist already.
+string CACHE_NAME = Guid.NewGuid().ToString();
+const string KEY = "MyKey";
+const string VALUE = "MyData";
+const uint DEFAULT_TTL_SECONDS = 60;
+
+using (SimpleCacheClient client = new SimpleCacheClient(Configurations.Laptop.Latest, MOMENTO_AUTH_TOKEN, DEFAULT_TTL_SECONDS))
 {
-    class Program
+    Console.WriteLine($"Creating cache {CACHE_NAME}");
+    var createCacheResponse = await client.CreateCacheAsync(CACHE_NAME);
+    // Check the create response for an error and handle as appropriate.
+    if (createCacheResponse is CreateCacheResponse.Error createError)
     {
-        static readonly String MOMENTO_AUTH_TOKEN = Environment.GetEnvironmentVariable("MOMENTO_AUTH_TOKEN");
-        static readonly String CACHE_NAME = "cache";
-        static readonly String KEY = "MyKey";
-        static readonly String VALUE = "MyData";
-        static readonly uint DEFAULT_TTL_SECONDS = 60;
-
-        async static Task Main(string[] args)
+        if (createError.ErrorCode == MomentoErrorCode.LIMIT_EXCEEDED_ERROR)
+        {
+            Console.WriteLine("Error: cache limit exceeded. We need to talk to support@moentohq.com! Exiting.");
+        } else
         {
-            using SimpleCacheClient client = new SimpleCacheClient(Configurations.Laptop.Latest, MOMENTO_AUTH_TOKEN, DEFAULT_TTL_SECONDS);
-            var createCacheResult = await client.CreateCacheAsync(CACHE_NAME);
+            Console.WriteLine($"Error creating cache: {createError.Message}. Exiting.");
+        }
+        // Any error is considered fatal.
+        Environment.Exit(1);
+    }
+    // If there's already a cache by this name, we don't want to modify it.
+    if (createCacheResponse is CreateCacheResponse.CacheAlreadyExists)
+    {
+        Console.WriteLine($"Error: strangely, a cache with the name {CACHE_NAME} already exists. Exiting.");
+        Environment.Exit(1);
+    }
```

Comment:
maybe I'm misunderstanding how we handle `CacheAlreadyExists` but I thought we want to pass it instead of exiting the program?

Reply:
For the most part that's true. The intent of this specific example is a little different, though, and this check is preventing the sample code from writing into a preexisting cache. This is really just showing that the response type exists and can be processed and acted upon.

Reply:
I like the log message but I think that's enough to illustrate how they could handle it, I think I'd lean towards making this not-fatal too